File: metrics-examples/src/main/ruby/io/vertx/example/metrics/prometheus/prometheus_metrics_verticle.rb
require 'vertx-web/router'
metricRegistry = Java::ComCodahaleMetrics::SharedMetricRegistries.get_or_create("exported")
Java::IoPrometheusClient::CollectorRegistry::defaultRegistry.register(Java::IoPrometheusClientDropwizard::DropwizardExports.new(metricRegistry))

#Bind metrics handler to /metrics
router = VertxWeb::Router.router($vertx)
router.get("/metrics").handler(&Java::IoPrometheusClientVertx::MetricsHandler.new())

#Start httpserver on localhost:8080
$vertx.create_http_server().request_handler(&router.method(:accept)).listen(8080)

#Increase counter every second
$vertx.set_periodic(1000) { |e|
  metricRegistry.counter("testCounter").inc()
}

File: metrics-examples/src/main/js/io/vertx/example/metrics/prometheus/prometheus_metrics_verticle.js
var Router = require("vertx-web-js/router");
var metricRegistry = Java.type("com.codahale.metrics.SharedMetricRegistries").getOrCreate("exported");
Java.type("io.prometheus.client.CollectorRegistry").defaultRegistry.register(new (Java.type("io.prometheus.client.dropwizard.DropwizardExports"))(metricRegistry));

//Bind metrics handler to /metrics
var router = Router.router(vertx);
router.get("/metrics").handler(new (Java.type("io.prometheus.client.vertx.MetricsHandler"))());

//Start httpserver on localhost:8080
vertx.createHttpServer().requestHandler(router.accept).listen(8080);

//Increase counter every second
vertx.setPeriodic(1000, function (e) {
  metricRegistry.counter("testCounter").inc();
});
